synapse/handlers/sso.py

```diff
@@ -141,7 +141,8 @@ class UserAttributes:
     confirm_localpart: bool = False
     display_name: Optional[str] = None
     picture: Optional[str] = None
-    emails: Collection[str] = attr.Factory(list)
+    # mypy thinks these are incompatible for some reason.
+    emails: StrCollection = attr.Factory(list)  # type: ignore[assignment]
```

Comment:
What was the error, out of interest?

Reply:
Either that or it can't figure out that `_T` is meant to be `str`.

(Not fully sure how attrs support in mypy works. IIRC I think they have an internal mypy plugin dedicated to attrs.)